docs/contributing/making-a-pull-request.md

```diff
@@ -0,0 +1,450 @@
+
+9. pull the changes to its master branch.
+
+## Steps
+
+Now that the overall process is outlined, here are specific instructions and
+tips. There are many choices to be made when describing a process for
+contributing to a project via a pull request. In the following, we assume that
+you are working with the Git command-line tools. For most alternatives (such as
+using IDEs or using functionality provided through the GitHub web interface),
+the translation from the command-line instructions should be simple enough.  We
+will add notes only where really necessary to keep the complexity of this to a
+reasonable level.
+
+### Forking the repository
+
+To make changes to Material for MkDocs, you would first fork one of its
+repositories on GitHub. This is so that you have a repository on GitHub that
+you can push changes to (only maintainers and collaborators have write access
+to the original repositories).
+
+Fork the [repository for the public version] if you want to make changes to
+code that is in the public version or if you want to make changes to the
+documentation. Please make sure to change the name of the repository by
+appending `-fork` to that people who come across it know that they have found a
+temporary fork rather then the original or a permanent fork of the project.
+You may want to add a description that clarifies what the repository is for.
+
+[repository for the public version]: https://github.com/squidfunk/mkdocs-material
+
+To make changes to functionality available only within the Insiders version,
+fork [the Insiders repository]. Note that the fork will be a private repository.
+Please respect the [terms of the Insiders program] and the spirit of the
+Sponsorware approach used to maintain and develop Material for MkDocs.
+
+[the Insiders repository]: https://github.com/squidfunk/mkdocs-material-insiders/
+[terms of the Insiders program]: http://localhost:8000/mkdocs-material/insiders/faq/sponsoring/#licensing
+
+### Setting up a development environment
+
+From this point onwards, please follow the [instructions for setting up the
+development environment]. They will take you through the process of setting up
+an environment in which you can make changes and review/test them.
+
+[instructions for setting up the development environment]: ../customization.md#environment-setup
+
+### Making changes
+
+When you make changes to the code or the documentation please follow the
+established style used in the project. Doing so increases readability and
+also helps with making diffs easier to read for those who will review the pull
+request. Avoid making any large-scale style changes such as asking your IDE
+to re-format all code.
+
+Study the code that you are modifying well to ensure that you fully understand
+how it works before you try to change it. This will not only help you solve the
+problem you are trying to address but also minimize the risks of creating
+unintended side effects.
+
+!!! tip "Linters"
```

Comment:
On part of my brain was already aware of this, I think. Question, though: how do they manifest themselves? Do they produce their output in the window that runs `npm start`? I usually minimize it as soon as it is running. 

For people using VScode, should they be installing specific plugins and do we want to point to those? I do note there is a setting in `.vscode` for stylelint and there are config files for ESLint in the project root. Not sure what else is needed, though? A related question would be about `.editorconfig` and how it might interact with VSCode user settings. People might be surprised by the editor behaviour if their editor honours `.editorconfig` and might be tempted to override the settings if they do not know where they come from.

Perhaps this is too much to bite off at the point and we should just remove the mention of linters and stuff for now?

Reply:
Drop it for now. We already talked about re-evaluating code-formatters for easier collaboration, so let's tackle that in the future instead of pointing out that we're not quite there. It hasn't been a huge deal in the past.